### ISSUE TITLE
Nearest departures ignores some stops

### ISSUE THREAD

**Issue by mikaelbr** [AtB-AS/atb-bff#16]
Looks like some stops are ignored when searching for nearest departures. For instance, when searching from stop `Sorgenfri`, none of the actual departures from that stop is included, but the stops around is.

See example:

![image](https://user-images.githubusercontent.com/606374/77436766-2b05e880-6de4-11ea-9cd6-b249a6c121ed.png)

I haven't looked into this much yet, just noting it here now as I found it.

I get that we probably need to be a bit more sophisticated with nearby at some point (both UI and business logic wise). Maybe it's due to me using the nearest endpoint and should use regular departure endpoints, though, but I think it still should show up?

**Comment by torfjor** [comment]
Could it simply be that there are very few departures from Sorgenfri?

AFAIK only bus 24 stops there.

Querying for departures from a place near Prinsens gate seems to give the expected results.

**Comment by mikaelbr** [comment]
For some reason I uploaded the wrong image. Should have been these:

![image](https://user-images.githubusercontent.com/606374/77441174-6b1b9a00-6de9-11ea-82c2-1932ef4df4bf.png)


![image](https://user-images.githubusercontent.com/606374/77441206-74a50200-6de9-11ea-93da-2e66c086b33a.png)

As you see on the second image here, the surrounding stops are included but not Sorgenfri. 24 stops on Sorgenfri, Valgrinvegen and Anton Grevskotts.

**Comment by torfjor** [comment]
I see. The query does the following:

Given `point`,
1: Find all stops around point 
2: Find all departures from stops
3: Flatten the array of departures and sort them by departure time.
4: Reduce right, keeping only departures with a unique serviceJourneyID.

So only the first departure for a given serviceJourneyID will be listed, in this example the one from Anton Grevskotts veg.

I guess this can be problematic in some cases, like the one you listed.

Maybe we should sort the departures by distance to `point` instead?

**Comment by mikaelbr** [comment]
Yeah I was about to say: maybe we should introduce haversine here

**Comment by torfjor** [comment]
I just pushed an update that sorts on distance to `point`.

**Comment by mikaelbr** [comment]
FYI, related change: 9787558e0ce10e044126cc1bfcb63f7e3248af2c

---

Hm. Checking this out again now that it is deployed. Now it seems odd that departure time is random down the list.

I was imagining something like this:

1. Find all stops around point
2. Find departures from stops
3. Reduce/flatMap departures with unique service journey ID selecting the departures which are closest. (haversine here). 
4. Sort by departure time.

**Comment by torfjor** [comment]
It's sorted by `expectedDepartureTime` now (albeit in wrong order :grin:). Which property is used in the label in the screenshot?

**Comment by mikaelbr** [comment]
I saw a discussion on this around realtime, and I belive it is `aimed`, or am I wrong, @jleknes ?

**Comment by torfjor** [comment]
From a recent query:

```
{
...
    "aimedArrivalTime": "2020-03-25T13:44:00+0100",
    "aimedDepartureTime": "2020-03-25T13:44:00+0100"
...
    "expectedDepartureTime": "2020-03-25T13:44:00+0100",
    "expectedArrivalTime": "2020-03-25T13:43:23+0100",
...
    "realtime": true,
...

}
```

**Comment by mikaelbr** [comment]
Yeah, doesn't that confirm that `aimedXX` is what is what we should use? At the very least, that is what I use in the UI now 😁

**Comment by mikaelbr** [comment]
Given any more thoughts on this, @torfjor ?

**Comment by torfjor** [comment]
@mikaelbr Maybe I'm missing something here, but `aimedXXX` is always rounded to a whole minute, ie. timetable departures.
